### PR TITLE
Update `MCM` related CRDs

### PR DESCRIPTION
**Pull request by rishabh-11**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
This PR updates the MCM related crds (machine, machinedeployment and machineset) to the latest ones. 

**Which issue(s) this PR fixes** 
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by rishabh-11** [comment]
cc @himanshu-kun

File: pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml [hunk 1]
```diff
@@ -4,8 +4,7 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.9.2
-  labels:
-    gardener.cloud/deletion-protected: "true"
```
Comment (ialidzhikov) [inline pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml]: Revert this and similar removals of the `gardener.cloud/deletion-protected` label.

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml (hunk 1 above)
Comment (rishabh-11) [reply pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml]: addressed in [41b62fc](https://github.com/gardener/gardener/pull/8491/commits/41b62fcdfc837b1b50815a56137615854350396b)

**Comment by rishabh-11** [comment]
make retest-required

File: pkg/component/machinecontrollermanager/templates/crd-machinesets.tpl.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ metadata:
     controller-gen.kubebuilder.io/version: v0.9.2
   labels:
     gardener.cloud/deletion-protected: "true"
+  creationTimestamp: null
```
Comment (himanshu-kun) [inline pkg/component/machinecontrollermanager/templates/crd-machinesets.tpl.yaml]:
```suggestion
```

File: pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ metadata:
     controller-gen.kubebuilder.io/version: v0.9.2
   labels:
     gardener.cloud/deletion-protected: "true"
+  creationTimestamp: null
```
Comment (himanshu-kun) [inline pkg/component/machinecontrollermanager/templates/crd-machinedeployments.tpl.yaml]:
```suggestion
```

File: pkg/component/machinecontrollermanager/templates/crd-machines.tpl.yaml [hunk 4]
```diff
@@ -6,6 +6,7 @@ metadata:
     controller-gen.kubebuilder.io/version: v0.9.2
   labels:
     gardener.cloud/deletion-protected: "true"
+  creationTimestamp: null
```
Comment (himanshu-kun) [inline pkg/component/machinecontrollermanager/templates/crd-machines.tpl.yaml]:
```suggestion
```

**Review by himanshu-kun — APPROVED** [review]
/lgtm , minor suggestions made

**Review by himanshu-kun — COMMENTED** [review]
/lgtm from my side

**Comment by ialidzhikov** [comment]
@rishabh-11 @himanshu-kun I pushed d0c25f427f8ff7a4ca1effb30eff55e7add77d19 to add a source information from where the MCM CRDs are actually maintained. This is similar to the etcd-druid CRDs (ref https://github.com/gardener/gardener/blob/ebda2b53e4b4e8893bfbbbb6510a4654c93ffdb7/pkg/component/etcd/crds/templates/crd-druid.gardener.cloud_etcds.yaml#L1) and VPA CRDs (https://github.com/gardener/gardener/blob/498403ac7be3f528d51e3647882155414f22289e/pkg/component/vpa/templates/crd-verticalpodautoscalers.tpl.yaml#L2).

**Comment by ialidzhikov** [comment]
That's the reason for the test failures - see https://github.com/gardener/gardener/pull/7599#issuecomment-1458311520. 
I think we have to first update out machine-controller-manager-provider-local to vendor newer MCM and to no longer rely on the `.status.node` field.

**Comment by rfranzke** [comment]
Please adapt https://github.com/gardener/gardener/blob/32c069f46a072f492e4e14c335d63337cb419da8/pkg/provider-local/imagevector/images.yaml#L2-L9 now that https://github.com/gardener/machine-controller-manager-provider-local/pull/18 got merged (new image is `ghcr.io/gardener/machine-controller-manager-provider-local:sha-8e34e10`).

**Pull request merged**